components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -148,7 +148,11 @@ public void buildResourceAccessControlData() {
             List<AuthenticationHandler> allAvailableAuthHandlers =
                     AuthenticationServiceHolder.getInstance().getAuthenticationHandlers();
             for (AuthenticationHandler handler : allAvailableAuthHandlers) {
-                allowedAuthHandlersList.add(handler.getName());
+                String handlerName = handler.getName();
+                if (handlerName.equals("BasicClientAuthenticationHandler")){
```

Comment:
Can we introduce a common constant for `BasicClientAuthenticationHandler`? Use the same constant in here also https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/151/files#diff-fc52a30d2c132741d459467538caa8e7d082087a342a84699a6f0add697b4f32R59